django/forms/boundfield.py

```diff
@@ -296,9 +296,10 @@ def build_widget_attrs(self, attrs, widget=None):
             not attrs.get("aria-describedby")
             and not widget.attrs.get("aria-describedby")
             and self.field.help_text
-            and self.id_for_label
+            and not self.use_fieldset
+            and self.auto_id
         ):
-            attrs["aria-describedby"] = f"{self.id_for_label}_helptext"
+            attrs["aria-describedby"] = f"{self.auto_id}_helptext"
```

Comment:
This was added in 966ecdd482167f3f6b08b00f484936c837751cb9 (Django 5.0). I'd recommend backporting this change to the Django 5.0 to avoid changes in Django 5.1 in recently added functionality.

Reply:
Sounds reasonable.

I was wondering if you may have triaged the other pr as a release blocker. I think it's not as we we've only recently agreed on a way forward for fields with related `<inputs>`.